### PR TITLE
Define IEquatable when already implemented

### PR DESCRIPTION
**Pull request by ie-zero**
All the classes amended have had implemented the IEquatable<T> interface already. This change simply add the interface at the class definition.

### CODE REVIEW

**Comment by ie-zero** [comment]
This is the follow up PR to #256 "Define IEquatable on Calendar and Occurrence classes"

File: net-core/Ical.Net/Ical.Net/DataTypes/PeriodList.cs
Comment (rianjs) [inline net-core/Ical.Net/Ical.Net/DataTypes/PeriodList.cs]: While you're here, let's make all `string.Equals` use `StringComparer.OrdinalIgnoreCase`.

File: net-core/Ical.Net/Ical.Net/DataTypes/RequestStatus.cs
Comment (rianjs) [inline net-core/Ical.Net/Ical.Net/DataTypes/RequestStatus.cs]: Let's use `OrdinalIgnoreCase`, here and elsewhere.

File: net-core/Ical.Net/Ical.Net/DataTypes/WeekDay.cs
Comment (rianjs) [inline net-core/Ical.Net/Ical.Net/DataTypes/WeekDay.cs]: Throwing an exception in a `CompareTo` method probably isn't a good idea. Consider the usage in a LINQ query:

```csharp
var orderedDays = weekdayCollection
    .OrderBy(w => w)
    .ToList();
```

That could `throw` in the event that WeekDay is null. I think being able to sort `null` is reasonable behavior: any non-null WeekDay has greater weight. Consider instead:

```csharp
public int CompareTo(WeekDay other)
{
    if (ReferenceEquals(this, other)) return 0;
    return ReferenceEquals(null, other)
        ? 1
        : DayOfWeek.CompareTo(other.DayOfWeek);
}
```

I'm not sure using `Offset` as a tie-breaker is necessary or desirable.

File: net-core/Ical.Net/Ical.Net/DataTypes/WeekDay.cs
Comment (rianjs) [inline net-core/Ical.Net/Ical.Net/DataTypes/WeekDay.cs]: Let's get rid of this `IComparable` implementation in favor of just `IComparable<T>`, so:
* Delete this method
* Delete `IComparable` from `IWeekDay` in v2, and delete `IComparable` from the top here in net-core.

**Comment by rianjs** [comment]
High-level thoughts:

* Let's delete all the non-generic `IComparable` and `IEquatable`s that exist, and use their generic equivalents instead: `IComparable<T>` and `IEquatable<T>`
* When you delete the non-generic interfaces, place the generic replacement on the object itself instead of the interface like you have done in this pull request. For example: `WeekDay` should implement `IComparable<WeekDay>`, and delete `IComparable` from `IWeekDay`. (I'm trying to simplify interface dependencies, which are out of control.)
* `IComparable<T>` implementations should allow null values without throwing exceptions. [This StackOverflow thread](http://stackoverflow.com/questions/8642080/icomparable-behaviour-for-null-arguments) explains what I mean.
* Where there are `string.Equals` calls, let's make sure they all specify `StringComparer.OrdinalIgnoreCase`. [Case-insensitive is OK for properties and most other component values, because RFC-5545 doesn't specify case sensitivity in most cases](https://tools.ietf.org/html/rfc5545#section-2).

**Comment by ie-zero** [comment]
Thank you Rian (@rianjs) for the code review and helpful feedback. The additional PRs, I believe, implement all the above points.

Please let me know about any further improvements that you might see fit.

File: net-core/Ical.Net/Ical.Net/Components/UniqueComponent.cs
Comment (rianjs) [inline net-core/Ical.Net/Ical.Net/Components/UniqueComponent.cs:112]: * Using the `Uid` as the basis for equality isn't consistent with the rest of the library. (I know that's the current implementation -- which is wrong, so not your fault.)
* This method should delegate to `RecurringComponent.Equals()` in the event that the element is recurring, and call its base class in the event that the element is NOT recurring. I don't think there's anything inside `UniqueComponent` that's useful for determining equality or hashing. 

`RecurringComponent` should know how to do a reasonable comparison and hash, but I think it's missing the `IEquatable<T>`, and isn't defining an `Equals` method, so we have nothing to delegate to. That will need to be implemented.

If you want, you can add the `IEquatable<T>` to the `RecurringComponent`, and define an `Equals` function. I think it will require some thought for both `GetHashCode` and `Equals`. If you don't want to do that, or would prefer to do it as a separate pull request, you can remove the `IEquatable` from this class, and add it back in with the changes to RecurringComponent. (Or I'll do it myself.)

Actually, I think that work should be done as a separate ticket, because it's a little more significant, and needs to be done carefully.

File: v2/ical.NET/Components/UniqueComponent.cs
Comment (rianjs) [inline v2/ical.NET/Components/UniqueComponent.cs:112]: (Same comment here,)

**Comment by ie-zero** [comment]
I also think it is better for this work to be done under a different PR. 

With this PR we are in at the excact same state as before (no better or worst) so, if you are happy, we can merge it in and open a ticket to rework the Equal() in RecurringComponent and UniqueComponent classes.

**Review by rianjs — APPROVED** [review]
Can you merge this after I submit my approval, or do I need to tweak the repo settings?

**Comment by rianjs** [comment]
Oh, two last, easy things to tack onto this PR:

* Please open the two nuspec files, and bump the minor revision number
* Please update `readme.md` to summarize the changes you made. Something like "All calendar components now explicitly implement `IEquatable<T>`", should be sufficient.

After that, I'll build and publish an updated nuget package.

**Comment by ie-zero** [comment]
I never pushed changes directly into the original repository. Sorry to be a pain but would it possible to provide me with some guidance or merge it yourself please?

**Pull request closed without merging**
